content/en/docs/plugins/developers_guide.md

```diff
@@ -0,0 +1,1081 @@
+
+A question to ask when deciding what to put in an event is "if this were written to a `.scap` capture file and reread, would this plugin return the same values for fields as it did when the events were first generated?".
+
+### Plugin Authoring Lifecycle
+
+Here are some considerations to keep in mind when releasing the initial version of a new plugin and when releasing updated versions of the plugin.
+
+#### Initial Version
+
+For source plugins, make sure the event source is distinct, or if the same as existing plugins, that the saved payload is identical. In most cases, each source plugin should define a new event source.
+
+For extractor plugins, if the plugin exports a set of compatible sources, make sure you have tested with capture files generated by each source plugin to ensure that your extractor plugin can read event payloads without errors/crashing. If the plugin does *not* export a set of compatible sources (meaning that it potentially handles every kind of event), your plugin must be very resilient. It will potentially be handed arbitrary binary data from other plugins.
+
+Register this plugin by submitting a PR to [falcosecurity/plugins](https://github.com/falcosecurity/plugins) to update [PLUGINS-REGISTRY.md](https://github.com/falcosecurity/plugins/blob/master/plugins/PLUGINS-REGISTRY.md). This will give an official Plugin ID that can be safely used in capture files, etc., without overlapping with other plugins. It also lets others know that a new plugin is available!
+
+#### Updates
+
+Every new release of a plugin should update the plugin's version number. Following semver conventions, the patch number should always be updated, the minor number should be updated when new fields are added, and the major number should be updated whenever any field is modified/removed or the semantics of a given field changes.
+
+With every release, you should check for an updated Plugin API Version and if needed, update the plugin to conform to the new API. Remember that a plugin and framework are considered be compatible if their major versions are the same.
+
+With each new release, make sure the contact information provided by the plugin is up-to-date.
+
+## Go Plugin SDK Walkthrough
+The [Go SDK](https://github.com/falcosecurity/plugin-sdk-go) provides prebuilt constructs and definitions that help developing plugins by abstracting all the complexities related to the bridging between the C and the Go runtimes. The Go SDK takes care of satisfying all the plugin framework requirements without having to deal with the low-level details, by also optimizing the most critical code paths.
+
+This section documents the Go SDK at a high-level, please refer to the [official Go SDK documentation](https://github.com/falcosecurity/plugin-sdk-go) for deeper details.
+
```

Comment:
I know I've mentioned this in other places like the godoc docs, and you kind of mention this in the next paragraph, but I'd suggest mentioning here that a sdk user can either choose from a lower level set of abstractions or just take the streamlined set of packages, where the "easy" route just involves importing a few packages. To me this will help a new reader avoid being overwhelmed by the packages and know they can just choose a streamlined interface, if they want.